report-ng/app/src/services/common-models.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Testerra
+ *
+ * (C) 2023, Selina Natschke, Deutsche Telekom MMS GmbH, Deutsche Telekom AG
+ *
+ * Deutsche Telekom AG and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export enum ChipType {
```

Comment:
When using string enums instead of numeric, you can have a one-by-one matching to the query parameters.
```typescript
export enum FilterType {
    CLASS="class",
    CUSTOM_TEXT="q",
}
```

It's better to refactor that to a generic filter type model like:
```typescript
type Filter = {
    type: FilterType,
    name: str,
    label: str,
}
```
and introduce all supported filters like that:
```typescript
const filters: Filter[] = [
    {
        "type": FilterType.CLASS,
        "name": "Test class",
        "label": "Search for Java test class names",
    },
    {
        "type": FilterType.CUSTOM_TEXT,
        "name": "Custom Text",
        "label": "Search text",
}
```

The `FilterChip` represents the view of this model and contains more properties like `class` and `label`.
```typescript
type FilterChip = {
    element: HTMLElement,
    class: str,
    filter: Filter,
    label: str
}
```
Then you can easily access all of its properties:
```typescript
this.queryParams[filter.type] = "MyClass"
delete this.queryParams[chip.filter.type]
```
That results in less complex and better readable code.
```html
<md-chip class.bind="chip.class` class="mr1"
mdc-tooltip="chip.label?chip.label:chip.filter.label">
<span>${chip.filter.name}</span>
```